tests/unit/db/account_tests.py

```diff
@@ -0,0 +1,600 @@
+            self.client.connect()
+            # Default returns None
+            self.assertIsNone(self.client.get('no_such_db', remote=True))
+            # Creates the database remotely and ensure that
+            # it is not in the client database local cache
+            db = self.client.create_database('unittestdb')
+            del self.client['unittestdb']
+            self.assertIsNone(self.client.get('unittestdb'))
+            # Retrieve the database object from the server
+            self.assertEqual(self.client.get('unittestdb', remote=True), db)
+        finally:
+            self.client.delete_database('unittestdb')
+            self.client.disconnect()
+
+    def test_set_non_db_value_via_setitem(self):
+        """
+        Test raising exception when value is not a database object
+        """
+        try:
+            self.client.connect()
+            self.client['unittestdb'] = 'This is not a database object'
+            self.fail('Above statement should raise a CloudantException')
+        except CloudantException, err:
+            self.assertEqual(str(err), 'Value must be set to a Database object')
+        finally:
+            self.client.disconnect()
+
+    def test_local_set_db_value_via_setitem(self):
+        """
+        Test setting a database object to the local database cache
+        """
+        try:
+            self.client.connect()
+            db = self.client._DATABASE_CLASS(self.client, 'unittestdb_local')
+            # Value is set in the local database cache but not on the server
+            self.client['unittestdb_local'] = db
+            self.assertEqual(self.client.get('unittestdb_local'), db)
+            self.assertFalse(db.exists())
+        finally:
+            self.client.disconnect()
+
+    def test_create_db_via_setitem(self):
+        """
+        Test creating a database remotely using __setitem__
+        """
+        try:
+            self.client.connect()
+            db = self.client._DATABASE_CLASS(self.client, 'unittestdb')
+            self.client.__setitem__('unittestdb', db, remote=True)
+            self.assertTrue(db.exists())
+        finally:
+            self.client.delete_database('unittestdb')
+            self.client.disconnect()
+
+@unittest.skipUnless(RUN_CLOUDANT_TESTS, 'Skipping Cloudant Account tests')
+class CloudantAccountTests(CouchDBAccountTests):
+
+    @classmethod
+    def setUpClass(self):
+        """
+        For the Cloudant tests we expect an account to already be created.
+        So we override the CouchDBAccountTests setUpClass and do nothing.
+        """
+        pass
+
+    @classmethod
+    def tearDownClass(self):
+        """
+        For the Cloudant tests we expect an account to already be created.
+        So we override the CouchDBAccountTests tearDownClass and do nothing.
+        """
+        pass
+
+    def setUp(self):
+        """
+        Set up test attributes for Cloudant Account tests
+        """
+        self.http = 'https'
```

Comment:
This should probably be switchable to http from env vars/command line - it's useful to troubleshoot network traffic with wireshark or similar tools and this can't be done with https.